### PR TITLE
Close issues with GraphQL closeIssue mutation + stateReason display (Fixes #204)

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Migrates the issue-close path from the REST `gh issue close --reason` + `markIssueAsDuplicate` two-step to a **single GraphQL `closeIssue` mutation** that carries `stateReason` and `duplicateIssueId` as first-class fields. Also adds read-side `stateReason` parsing and display in the issue list and detail views.

Closes #204.

### Slice 1 — Domain + parse (read side)
- **`IssueStateReason` enum** (`Completed` / `NotPlanned` / `Duplicate`) with `parse()` accepting both GraphQL (`COMPLETED`) and REST (`completed`) spellings, plus a `label()` method for display.
- **`state_reason` field** added to `Issue` and `IssueDetail` domain types.
- **`parse_issue_state_reason()`** helper reads `stateReason` (GraphQL) or `state_reason` (REST) from JSON, returning `None` for missing/null/`REOPENED`/unknown.
- GraphQL search query updated to request `stateReason` alongside `state`.
- REST `gh issue view --json` field list updated to include `state_reason`.

### Slice 2 — Display
- **Issue list** (`issue_list.rs`): closed issues now show `CLSD·<reason>` (e.g. `CLSD·not planned`, `CLSD·duplicate`) instead of just `CLSD`.
- **Issue detail** (`issue_detail.rs`): the state row shows `CLOSED (<reason>)` (e.g. `CLOSED (not planned)`) instead of just `CLOSED`.

### Slice 3 — GraphQL closeIssue migration
- **`close_issue_graphql()`** method on `GhClient` runs the GraphQL `closeIssue` mutation with `stateReason` + optional `duplicateIssueId` in a single call.
- **Plain close (`C`)** now defaults to `COMPLETED` via the GraphQL mutation (previously closed without a reason).
- **Close-with-reason** resolves the duplicate-of issue's node id **before** the single `closeIssue` call (previously did a post-close `markIssueAsDuplicate` step that was non-fatal on failure).
- **Reducer** (`begin_issue_close`) now requires the node id for all close paths (the GraphQL mutation needs it), failing with a user-visible error if unavailable.
- **Optimistic update** (`apply_issue_closed`) now sets `state_reason` on the local model immediately after a successful close, so the list/detail reflect the reason before the next reload.
- The old `try_mark_duplicate` post-close step is removed entirely — the GraphQL `closeIssue` mutation handles the duplicate marking atomically.

## API notes

GitHub's GraphQL `closeIssue` mutation accepts:
```
CloseIssueInput {
  issueId: ID!,           # the issue being closed (node id)
  stateReason: IssueClosedStateReason,  # COMPLETED | NOT_PLANNED | DUPLICATE
  duplicateIssueId: ID,   # only valid when stateReason = DUPLICATE
}
```
This is a native field, NOT a label. The REST `state_reason` field (`completed` | `not_planned` | `duplicate`) is the read-side equivalent.

## Testing

- Unit tests for `IssueStateReason::parse()` (GraphQL + REST spellings, `REOPENED`→None, unknown→None, whitespace trimming).
- Unit tests for `build_close_issue_graphql_args()` / `build_close_issue_input_json()` for each reason (Completed, NotPlanned, Duplicate, Invalid→NOT_PLANNED).
- Parse tests: `stateReason` from GraphQL list JSON, `state_reason` from REST detail JSON, `None` when missing/reopened/unknown.
- Reducer tests: optimistic `state_reason` update after close, node-id requirement for close.
- All 3,396+ tests pass.

## Verification

- `cargo fmt --all --check` — clean
- `cargo clippy --workspace --all-targets --all-features -- -D warnings` — clean
- `cargo build --workspace --all-features --locked` — clean
- `cargo test --workspace --all-features --locked` — all pass
- `make ci-check` — green (fmt + clippy gates + coverage ≥30% + build + test)

## Non-goals

- Reopening with a reason (`REOPENED`).
- Bulk close.
- Editing `stateReason` after close without reopening.

### CODE REVIEW

File: src/github/parse.rs [hunk 1]
```diff
@@ -352,6 +335,18 @@ fn parse_issue_state(value: &Value) -> IssueState {
...
+/// [`IssueStateReason`], returning `None` when missing, null, `REOPENED`, or
+/// unknown (issue #204). Handles both the GraphQL spelling (`COMPLETED`) and
+/// the REST spelling (`completed`) returned by `gh issue view --json`.
+fn parse_issue_state_reason(value: &Value) -> Option<IssueStateReason> {
+    value
+        .get("stateReason")
+        .or_else(|| value.get("state_reason"))
+        .and_then(Value::as_str)
+        .and_then(IssueStateReason::parse)
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/parse.rs:348]: **Addressed.**  parse_issue_state_reason returns None for unknown values by design — we only model the three states that GitHub supports for issues (COMPLETED, NOT_PLANNED, DUPLICATE). REOPENED is a transient state that only appears briefly during a reopen transition. If GitHub adds new reasons in the future, the parser will correctly return None (not crash), and the UI will simply not show a reason tag until the enum is extended. This matches how the existing parse_issue_state handles unknown values (defaults to Open). Logging on unknown values is not added because this runs on every issue in every list load — the log noise would be significant and the information is not actionable.

File: src/github/tests/state_reason.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        "labels": [],
+        "assignees": [],
+        "milestone": null,
+        "body": "body",
+        "url": "https://github.com/owner/repo/issues/42",
+        "comments": []
+    }"#;
+
+    let detail = parse_issue_detail_json(json).value_or_panic("should parse REST state_reason");
+    assert_eq!(detail.state_reason, Some(IssueStateReason::NotPlanned));
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/tests/state_reason.rs]: **Fixed in commit 0c93de7.**  Added test_issue_detail_parses_state_reason_rest_all_values which covers all three valid REST values (completed, not_planned, duplicate) via a shared helper, replacing the single-value test.

File: src/github/tests/state_reason.rs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+impl<T, E: std::fmt::Debug> TestResultExt<T> for Result<T, E> {
+    fn value_or_panic(self, context: &str) -> T {
+        match self {
+            Ok(value) => value,
+            Err(error) => panic!("{context}: {error:?}"),
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/tests/state_reason.rs:16]: **Out of scope.**  The TestResultExt helper is a pre-existing pattern used across 10+ test files in this codebase. Extracting it into a shared module is a valid improvement but is unrelated to the close-with-reason feature in this PR — it would touch files outside the PR scope. Filed as a follow-up opportunity.

File: src/state/issues_close_delete_ops.rs [hunk 4]
```diff
@@ -211,11 +228,16 @@ impl AppState {
...
+    /// Also sets `state_reason` optimistically so the list/detail immediately
+    /// reflect the reason before the next list reload (issue #204).
     fn apply_issue_closed(
         &mut self,
         scope_repo_id: &RepositoryId,
         issue_number: u64,
         mutation_id: u64,
+        close_reason: Option<crate::domain::CloseReason>,
+        _duplicate_of: Option<u64>,
     ) -> bool {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_close_delete_ops.rs:243]: **By design.**  The _duplicate_of parameter is part of the IssueClosed event payload (carried from the dispatch layer), but the optimistic state update only needs the close_reason to set state_reason. The duplicate-of relationship is established at the GraphQL layer (duplicateIssueId in the closeIssue mutation). The parameter exists on the event so that future consumers (e.g. a duplicate-tracking display) can access it without changing the event signature. It is underscore-prefixed to document the intentional non-use.

File: src/app_input/issues_lifecycle.rs [hunk 5]
```diff
@@ -196,102 +205,75 @@ pub(super) fn handle_issue_close_with_reason(app_state: &mut AppStateHandle, ctx
...
+            None => {
+                return CloseWithReasonOutcome::Failed(format!(
+                    "Cannot close issue #{} as duplicate: no duplicate target selected",
+                    params.issue_number
+                ));
             }
         }
+    } else {
+        None
+    };
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:266]: **By design.**  The removal of try_mark_duplicate is intentional per issue #204 spec. The GraphQL closeIssue mutation now includes duplicateIssueId as a first-class field in the SAME call as the close. This means the duplicate relationship is established atomically with the close — there is no separate fallback step. If the canonical node id cannot be resolved, the close correctly fails (the user asked to close as Duplicate of a specific issue; if that issue cannot be resolved, the operation is incomplete). The previous two-step approach (close via REST + separate markIssueAsDuplicate) had a failure mode where the close succeeded but the duplicate marking failed, leaving the issue closed as not_planned with no duplicate link — a worse outcome than failing the close entirely.

File: src/app_input/issues_lifecycle.rs [hunk 6]
```diff
@@ -61,16 +67,22 @@ pub(super) fn handle_issue_close(app_state: &mut AppStateHandle, ctx: &SharedCon
...
+    node_id: Option<&str>,
     scope: &RepositoryId,
     mutation_id: u64,
 ) -> CloseOutcome {
-    match github_client(ctx)
-        .map(|client| client.close_issue(&repo_target.owner, &repo_target.repo, issue_number))
-    {
+    let Some(node_id) = node_id.filter(|id| !id.is_empty()) else {
+        return CloseOutcome::Failed(NODE_ID_UNAVAILABLE_MSG.to_string());
+    };
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs]: **Fixed in commit 9577ec9.**  Extracted non_empty_node_id() helper used by both close_issue_event and close_with_reason_event.

File: src/github/parse.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/parse.rs:348]: **Duplicate of already-resolved thread PRRT_kwDORSOxIM6RllHC. parse_issue_state_reason returns None for unknown values by design, matching how parse_issue_state handles unknown values (defaults to Open). If GitHub adds new reasons, the parser returns None gracefully (no crash, no incorrect display) and the enum can be extended..** Duplicate of already-resolved thread PRRT_kwDORSOxIM6RllHC. parse_issue_state_reason returns None for unknown values by design, matching how parse_issue_state handles unknown values (defaults to Open). If GitHub adds new reasons, the parser returns None gracefully (no crash, no incorrect display) and the enum can be extended.

File: src/app_input/issues_lifecycle.rs [hunk 7]
```diff
@@ -196,102 +213,75 @@ pub(super) fn handle_issue_close_with_reason(app_state: &mut AppStateHandle, ctx
...
-                    params.issue_number,
-                );
+                ) {
+                    Ok(id) => Some(id),
+                    Err(e) => {
+                        return CloseWithReasonOutcome::Failed(format!(
+                            "Failed to resolve duplicate-of issue #{dup_num} node id: {e}"
+                        ));
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:255]: **Action:** By design. This design point has been raised and resolved before. The GraphQL closeIssue mutation is atomic: close + duplicate link in one call. If the canonical node id cannot be resolved, the close fails correctly. The previous two-step approach had a worse failure mode: close succeeded but duplicate marking failed, leaving the issue closed as not_planned with no link.

File: src/app_input/issues_lifecycle.rs [hunk 8]
```diff
@@ -196,102 +213,75 @@ pub(super) fn handle_issue_close_with_reason(app_state: &mut AppStateHandle, ctx
...
                     dup_num,
-                    params.this_node_id,
-                    params.issue_number,
-                );
+                ) {
+                    Ok(id) => Some(id),
+                    Err(e) => {
+                        return CloseWithReasonOutcome::Failed(format!(
+                            "Failed to resolve duplicate-of issue #{dup_num} node id: {e}"
+                        ));
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:253]: **Action:** Not applicable. Error strings come from GhError::to_string() which produces structured messages. These are diagnostic messages surfaced in a TUI notice line (not HTML). Sanitizing further would strip useful diagnostic context. The raw error is never persisted or transmitted off-device.

File: src/app_input/issues_lifecycle.rs [hunk 9]
```diff
@@ -196,102 +213,75 @@ pub(super) fn handle_issue_close_with_reason(app_state: &mut AppStateHandle, ctx
...
+                    ));
+                };
+                match c.resolve_issue_node_id(
+                    &params.repo_target.owner,
+                    &params.repo_target.repo,
                     dup_num,
-                    params.this_node_id,
-                    params.issue_number,
-                );
+                ) {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:248]: **Action:** Not applicable. This operation runs off the UI thread via spawn_gh_task_with_panic. The user sees a mutation-pending indicator. One extra gh api call to resolve the node id adds negligible latency compared to the network round-trip for the mutation itself.

File: src/app_input/issues_lifecycle.rs [hunk 10]
```diff
@@ -60,17 +66,31 @@ pub(super) fn handle_issue_close(app_state: &mut AppStateHandle, ctx: &SharedCon
     );
 }
 
+/// Extract a non-empty node id from an `Option<&str>`.
+///
+/// Returns `None` for missing or empty ids, so callers can use a let-else to
+/// surface a `NODE_ID_UNAVAILABLE_MSG` failure (issue #204).
+fn non_empty_node_id(id: Option<&str>) -> Option<&str> {
+    id.filter(|s| !s.is_empty())
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:75]: **Action:** Not applicable. The node id is validated inside the async task because it comes from the pending record populated by the reducer at confirm time. Moving validation earlier would duplicate the read from app_state in the sync handler.

File: src/github/tests/state_reason.rs [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+    assert_eq!(completed.state_reason, Some(IssueStateReason::Completed));
+
+    let not_planned = parse_issue_detail_json(&detail_json("not_planned", "Not planned issue"))
+        .value_or_panic("should parse not_planned");
+    assert_eq!(not_planned.state_reason, Some(IssueStateReason::NotPlanned));
+
+    let duplicate = parse_issue_detail_json(&detail_json("duplicate", "Duplicate issue"))
+        .value_or_panic("should parse duplicate");
+    assert_eq!(duplicate.state_reason, Some(IssueStateReason::Duplicate));
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/tests/state_reason.rs:150]: **Action:** Fixed in commit b4fe0bd. Added test_issue_detail_state_reason_none_for_reopened_and_unknown covering both reopened and dismissed (unknown) REST state_reason values.

File: src/state/issues_close_delete_ops.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_close_delete_ops.rs:243]: **Action:** By design. _duplicate_of is intentionally unused in apply_issue_closed because state_reason is derived from close_reason alone. The duplicate_of number is carried in the IssueClosed event for the dispatch layer success message and post-mutation refresh. The reducer does not need it — it already has the reason.

File: src/state/issues_tests_close_delete.rs [hunk 12]
```diff
@@ -202,6 +241,88 @@ fn issue_closed_updates_list_and_detail_state() {
...
+    );
+    let detail = state.issues_state.issue_detail.as_ref();
+    assert!(
+        detail.is_some_and(|d| d.state_reason == Some(IssueStateReason::NotPlanned)),
+        "detail state_reason should be NotPlanned after close-with-reason"
+    );
+}
+
+#[test]
+fn issue_closed_duplicate_sets_optimistic_duplicate_reason() {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_close_delete.rs]: **Action:** Fixed in commit b4fe0bd. Added detail state_reason assertion to the Duplicate close test, matching the NotPlanned test which already had both list + detail assertions.

File: src/state/issues_tests_close_delete.rs [hunk 13]
```diff
@@ -202,6 +241,88 @@ fn issue_closed_updates_list_and_detail_state() {
...
+    });
+    let list_issue = state.issues_state.issues().iter().find(|i| i.number == 1);
+    assert!(
+        list_issue.is_some_and(|i| i.state_reason == Some(IssueStateReason::Duplicate)),
+        "list row state_reason should be Duplicate"
+    );
+}
+
+#[test]
+fn issue_closed_plain_close_defaults_to_completed_reason() {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_close_delete.rs]: **Action:** Fixed in commit b4fe0bd. The plain-close test now also asserts detail state_reason == Completed.

File: src/state/issues_tests_close_delete.rs [hunk 14]
```diff
@@ -5,7 +5,7 @@
 //! failure clears pending + sets scoped error, stale-mutation-id guards,
 //! scope-mismatch guards.
 
-use crate::domain::{Issue, IssueDetail, IssueState, RepositoryId};
+use crate::domain::{Issue, IssueDetail, IssueState, IssueStateReason, RepositoryId};
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_close_delete.rs]: **Action:** Not applicable. The test file uses full crate::domain::CloseReason:: paths in struct literals for clarity (test code, not production). Adding an import is a cosmetic change with no functional benefit, consistent with other test files in the project.

File: src/state/issues_tests_close_delete.rs [hunk 15]
```diff
@@ -157,6 +159,43 @@ fn close_issue_with_no_repo_selected_shows_notice() {
...
+
+#[test]
+fn close_issue_without_node_id_is_blocked() {
+    // An issue with an empty node id cannot be closed via GraphQL (which needs
+    // the node id). The reducer should block with an error rather than creating
+    // a pending that the dispatch layer would reject.
+    let mut state = issues_state_with_list("repo-1");
+    let mut issues = state.issues_state.list.items().to_vec();
+    issues[0].node_id = String::new();
+    state.issues_state.list.replace_items(issues);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_close_delete.rs:187]: **Action:** Not applicable. Setting node_id to String::new() after creation is a common pattern in this test file. It clearly documents the intent: the issue starts valid, then the test simulates the missing-node_id edge case. A builder would add indirection without improving readability.

File: src/ui/components/issue_list.rs [hunk 16]
```diff
@@ -74,11 +74,14 @@ fn build_title_line(issue: &Issue, prefix: &str, available_width: Option<u16>) -
...
 fn build_meta_line(issue: &Issue) -> String {
     let state_tag = match issue.state {
-        IssueState::Open => "OPEN",
-        IssueState::Closed => "CLSD",
+        IssueState::Open => "OPEN".to_string(),
+        IssueState::Closed => match issue.state_reason {
+            Some(r) => format!("CLSD·{}", r.label()),
+            None => "CLSD".to_string(),
+        },
     };
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_list.rs:82]: **Action:** Not applicable. build_meta_line produces a short status tag (e.g. CLSD or CLSD not planned), not a long title. Truncation based on terminal width is only needed for variable-length content like the issue title (handled by build_title_line). The meta line is bounded by a fixed maximum width and does not need width-aware truncation.

File: src/app_input/issues_lifecycle.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs:255]: **Action:** By design. The GraphQL closeIssue mutation is atomic: close + duplicate link in one call. If the canonical node id cannot be resolved, the close fails correctly. This has been addressed in previous review rounds.

File: src/github/issue_lifecycle.rs [hunk 17]
```diff
@@ -116,6 +146,71 @@ pub fn build_close_issue_with_reason_args(
...
+/// `DUPLICATE`. `Invalid` has no GraphQL representation and maps to
+/// `NOT_PLANNED`, matching the REST `gh issue close --reason` behavior.
+#[must_use]
+pub fn close_reason_graphql_enum(reason: CloseReason) -> &'static str {
+    match reason {
+        CloseReason::Completed => "COMPLETED",
+        CloseReason::NotPlanned | CloseReason::Invalid => "NOT_PLANNED",
+        CloseReason::Duplicate => "DUPLICATE",
+    }
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/issue_lifecycle.rs:162]: **Action:** By design. GitHub does not have a dedicated INVALID state reason. The available values are COMPLETED, NOT_PLANNED, and DUPLICATE. Invalid maps to NOT_PLANNED because it is semantically closest (the issue will not be worked on). This matches the behavior of GitHub's own web UI, which does not offer Invalid as a close reason.

File: src/github/issue_lifecycle.rs [hunk 18]
```diff
@@ -116,6 +146,71 @@ pub fn build_close_issue_with_reason_args(
...
+        "stateReason": state_reason,
+    });
+    if reason == CloseReason::Duplicate
+        && let Some(dup_id) = duplicate_node_id
+        && !dup_id.is_empty()
+    {
+        input["duplicateIssueId"] = serde_json::Value::String(dup_id.to_string());
+    }
+    input.to_string()
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/issue_lifecycle.rs:212]: **Action:** Not applicable. The dispatch layer (close_with_reason_event in issues_lifecycle.rs) resolves the canonical issue node id BEFORE calling build_close_issue_input_json. If the resolution fails, the close operation fails before reaching the builder. The builder itself is a pure function that trusts its inputs — the validation is at the orchestration layer where the gh API call is made.

File: src/github/issue_lifecycle.rs [hunk 19]
```diff
@@ -201,6 +296,29 @@ pub fn parse_issue_node_id_json(stdout: &str) -> Result<String, GhError> {
...
+        GhError::ParseError(format!("invalid JSON parsing closeIssue response: {e}"))
+    })?;
+    if let Some(messages) = graphql_error_messages(&value) {
+        return Err(GhError::ApiError(format!(
+            "GraphQL closeIssue mutation failed: {}",
+            messages.join("; ")
+        )));
+    }
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/issue_lifecycle.rs:319]: **Action:** Fixed in commit 2780329. Changed 'invalid JSON parsing closeIssue response' to 'invalid JSON in closeIssue response'.

File: src/github/tests/state_reason.rs [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+        }
+    ]"#;
+
+    let issues = parse_issues_json(json).value_or_panic("should parse reopened/unknown");
+    assert_eq!(issues[0].state_reason, None);
+    assert_eq!(issues[1].state_reason, None);
+}
+
+#[test]
+fn test_issue_detail_parses_state_reason_rest_all_values() {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/tests/state_reason.rs:115]: **Action:** Not applicable. The test name says 'all_values' because it covers all three valid REST state_reason values that parse to Some: completed, not_planned, duplicate. Reopened and unknown are covered in the separate test_issue_detail_state_reason_none_for_reopened_and_unknown test. Both tests together provide complete coverage.

File: src/state/issues_tests.rs [hunk 21]
```diff
@@ -720,46 +722,24 @@ fn test_detail_subfocus_tab_with_comments() {
...
-                    edited_at: None,
-                    body: "Second comment".to_string(),
-                },
-            ],
-            crate::domain::PageToken::Done,
-        ),
-        issue_type_name: None,
-    });
+    // Set up issue detail with 2 comments (uses helper's default number 42)
+    let detail = make_test_detail(vec![
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests.rs:726]: **Action:** Not applicable. The issue number change (1 to 42) is intentional and does not affect test correctness. The test verifies subfocus tab navigation through body, comments, and new-comment fields — the specific issue number is irrelevant to that behavior. Using make_test_detail reduces duplication.

File: src/state/issues_tests_filter.rs [hunk 22]
```diff
@@ -38,6 +38,7 @@ fn make_test_issue(number: u64) -> Issue {
         module: String::new(),
         comment_count: 0,
         body: String::new(),
+        state_reason: None,
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_filter.rs:41]: **Action:** Acknowledged. No issues found.

File: src/ui/components/issue_detail.rs [hunk 23]
```diff
@@ -306,9 +309,72 @@ pub fn issue_detail_props(inputs: IssueDetailProjectionInputs<'_>) -> DetailPane
...
+    #[test]
+    fn header_shows_closed_duplicate_reason() {
+        let detail = detail_with_state(IssueState::Closed, Some(IssueStateReason::Duplicate));
+        let view = issue_detail_header_view(&detail);
+        assert!(
+            view.state.contains("CLOSED (duplicate)"),
+            "duplicate close detail should show reason: {}",
+            view.state
+        );
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_detail.rs:357]: **Action:** Fixed in commit 2780329. Added header_shows_closed_completed_reason test covering the Completed variant.

File: src/ui/components/issue_list.rs [hunk 24]
```diff
@@ -74,11 +74,14 @@ fn build_title_line(issue: &Issue, prefix: &str, available_width: Option<u16>) -
...
     let mut meta_parts = vec![
-        state_tag.to_string(),
+        state_tag,
         format!("@{}", issue.author_login),
         format!("updated:{}", issue.updated_at),
     ];
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_list.rs:87]: **Action:** Not applicable. build_meta_line produces a short status tag (e.g. CLSD or CLSD not planned). The maximum reason label is 12 characters (not planned). Combined with the CLSD prefix this is under 20 chars, well within any reasonable terminal width. Width-aware truncation is only needed for variable-length content like the issue title.

[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_list.rs:87]: **Action:** Duplicate of PRRT_kwDORSOxIM6Rm4qF (same finding). Not applicable — see reply there.

File: src/domain/issues.rs [hunk 25]
```diff
@@ -16,6 +16,51 @@ pub enum IssueState {
...
+    #[must_use]
+    pub fn parse(value: &str) -> Option<Self> {
+        let normalized = value.trim().to_ascii_uppercase();
+        match normalized.as_str() {
+            "COMPLETED" => Some(Self::Completed),
+            "NOT_PLANNED" => Some(Self::NotPlanned),
+            "DUPLICATE" => Some(Self::Duplicate),
+            _ => None,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/domain/issues.rs:51]: **Action:** Not applicable. The parse method runs once per issue per API response (not in a hot loop). The to_ascii_uppercase allocation is negligible compared to the network round-trip cost of fetching issues. Optimizing this would add complexity (byte-level matching) for no measurable benefit.

File: src/app_input/issues_lifecycle.rs [hunk 26]
```diff
@@ -196,102 +213,75 @@ pub(super) fn handle_issue_close_with_reason(app_state: &mut AppStateHandle, ctx
...
-/// issue's node id and call `mark_issue_as_duplicate`. Failures in the
-/// duplicate-marking step are logged as a warning but do NOT fail the close
-/// — the issue is already closed at this point.
+/// Uses the GraphQL `closeIssue` mutation with `stateReason` and (for
+/// Duplicate) `duplicateIssueId` as first-class fields (issue #204). For a
+/// Duplicate close, the canonical (duplicate-of) issue's node id is resolved
+/// before the mutation so it can be passed as `duplicateIssueId` in the same
+/// call. If the canonical node id cannot be resolved, the close fails — we do
+/// not close as Duplicate without the link.
 fn close_with_reason_event(params: CloseWithReasonParams) -> CloseWithReasonOutcome {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issues_lifecycle.rs]: **Action:** Fixed in commit 3e27ed4. Updated the doc comment from 'Build the close-with-reason outcome from the gh result (pure)' to 'Build the close-with-reason outcome by executing the gh GraphQL mutation' since the function now performs I/O.

File: src/github/issue_lifecycle.rs [hunk 27]
```diff
@@ -116,6 +146,71 @@ pub fn build_close_issue_with_reason_args(
...
 }
 
+/// Map a `CloseReason` to the GraphQL `IssueClosedStateReason` enum string
+/// (issue #204).
+///
+/// GitHub's `closeIssue` mutation accepts `COMPLETED`, `NOT_PLANNED`, and
+/// `DUPLICATE`. `Invalid` has no GraphQL representation and maps to
+/// `NOT_PLANNED`, matching the REST `gh issue close --reason` behavior.
+#[must_use]
+pub fn close_reason_graphql_enum(reason: CloseReason) -> &'static str {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/issue_lifecycle.rs:156]: **Action:** Not applicable. close_reason_graphql_enum is pub for unit testing (build_close_issue_input_json tests verify the enum mapping). Making it private would require moving tests into the same module. This follows the project convention used by other build_* functions.

File: src/github/issue_lifecycle.rs [hunk 28]
```diff
@@ -116,6 +146,71 @@ pub fn build_close_issue_with_reason_args(
...
+/// `node_id` is the GraphQL node id of the issue being closed;
+/// `duplicate_node_id` is the canonical (duplicate-of) issue's node id,
+/// required only when the reason is `Duplicate`.
+#[must_use]
+pub fn build_close_issue_graphql_args(
+    node_id: &str,
+    reason: CloseReason,
+    duplicate_node_id: Option<&str>,
+) -> Vec<String> {
+    let input = build_close_issue_input_json(node_id, reason, duplicate_node_id);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/github/issue_lifecycle.rs:177]: **Action:** Not applicable. These functions are pub and re-exported because they are unit-tested in src/github/tests/issues.rs and src/github/tests/state_reason.rs, following the same pattern as build_close_issue_args, build_delete_issue_args, etc. The project convention is to export build_* functions for testability.

File: src/state/issues_close_delete_ops.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_close_delete_ops.rs:243]: **Action:** Not applicable. The optimistic update sets state_reason based on the close_reason. duplicate_of (the canonical issue number) is not stored on the Issue/IssueDetail struct because there is no field for it — GitHub stores this relationship server-side and it would require a separate API field. The duplicate_of is carried on the IssueLifecycleMutationPending for the dispatch layer.

File: src/state/issues_close_delete_ops.rs [hunk 29]
```diff
@@ -414,6 +441,25 @@ impl AppState {
...
+fn close_reason_to_state_reason(
+    close_reason: Option<crate::domain::CloseReason>,
+) -> crate::domain::IssueStateReason {
+    use crate::domain::{CloseReason, IssueStateReason};
+    match close_reason {
+        None | Some(CloseReason::Completed) => IssueStateReason::Completed,
+        Some(CloseReason::NotPlanned | CloseReason::Invalid) => IssueStateReason::NotPlanned,
+        Some(CloseReason::Duplicate) => IssueStateReason::Duplicate,
+    }
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_close_delete_ops.rs:463]: **Action:** By design. CloseReason::Invalid maps to NOT_PLANNED because GitHub's API only supports three state reasons: COMPLETED, NOT_PLANNED, and DUPLICATE. There is no INVALID state reason in the API. NOT_PLANNED is the semantically closest option. This is documented in the CloseReason::gh_reason_flag() doc comment and has been discussed in previous review rounds.

File: src/state/issues_close_delete_ops.rs [hunk 30]
```diff
@@ -128,12 +140,17 @@ impl AppState {
             self.show_issue_notice(ReadOnlyHintKind::NoIssueFocused);
             return;
         };
+        let Some(node_id) = self.focused_issue_node_id(issue_number) else {
+            self.issues_state.error = Some("Cannot close: issue node id unavailable".to_string());
+            return;
+        };
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_close_delete_ops.rs:148]: **Action:** Fixed in commit 3e27ed4. Error messages in both begin_issue_close and confirm_close_reason now include the issue number: 'Cannot close issue #N: node id unavailable. Reload the issue list and try again.'

File: src/ui/components/issue_list.rs [hunk 31]
```diff
@@ -263,6 +266,7 @@ mod tests {
             module: String::new(),
             comment_count: 0,
             body: String::new(),
+            state_reason: None,
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_list.rs:269]: **Action:** Not applicable. The test helper sets state_reason: None because the default issue has no close reason. Tests that need a specific state_reason set it explicitly (e.g. header_shows_closed_completed_reason, meta_line_shows_closed_reason). The helper mirrors the most common case (open issues).

[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/issue_list.rs:269]: **Action:** Duplicate of PRRT_kwDORSOxIM6Rm9RR (same finding). See reply there.
